creator-node/src/components/replicaSet/URSMRegistrationComponentService.js

```diff
@@ -0,0 +1,168 @@
+   */
+  const spDetailsFromSPFactory = await audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceProviderDetails(
+    ownerWalletFromSPFactory
+  )
+  if (!spDetailsFromSPFactory.validBounds) {
+    throw new ErrorBadRequest('ServiceProvider for given spID is not within valid bounds on L1 ServiceProviderFactory')
+  }
+
+  /**
+   * Request node's up-to-date health info
+   *  - uses endpoint registered on L1 for spID
+   *  - passes `randomBytesToSign` string in request to check that response was signed for provided data
+   */
+  const randomBytesToSign = (await randomBytes(18)).toString()
+  let nodeHealthCheckResp = await axios({
+    baseURL: nodeEndpointFromSPFactory,
+    url: '/health_check',
+    method: 'get',
+    timeout: 1000,
+    params: {
+      randomBytesToSign
+    }
+  })
+  nodeHealthCheckResp = parseCNodeResponse(
+    nodeHealthCheckResp,
+    ['healthy', 'creatorNodeEndpoint', 'spID', 'spOwnerWallet', 'randomBytesToSign']
+  )
+
+  /**
+   * Confirm health check returns healthy and response data matches on-chain data
+   */
+  if (
+    !(nodeHealthCheckResp.healthy) ||
+    (nodeHealthCheckResp.creatorNodeEndpoint !== nodeEndpointFromSPFactory) ||
+    (nodeHealthCheckResp.spID !== spID) ||
+    ((nodeHealthCheckResp.spOwnerWallet).toLowerCase() !== ownerWalletFromSPFactory.toLowerCase())
+  ) {
+    throw new ErrorServerError(
+      `Content node health check response from endpoint ${nodeEndpointFromSPFactory} indicates unhealthy or misconfigured`
+    )
+  }
+
+  /**
+   * Confirm health check response was signed by delegate owner wallet registered on L1
+   *    for spID and includes `randomBytesToSign`
+   */
+  let {
+    timestamp: respTimestamp,
+    signature: respSignature,
+    ...responseData
+  } = nodeHealthCheckResp.rawResponse
+  if (signatureHasExpired(respTimestamp)) {
+    throw new ErrorBadRequest('Health check response signature has expired')
+  }
+  const responderWalletRecoveryObj = { ...responseData, randomBytesToSign, timestamp: respTimestamp }
+  recoveredDelegateOwnerWallet = (recoverWallet(responderWalletRecoveryObj, respSignature)).toLowerCase()
+  if (delegateOwnerWalletFromSPFactory !== recoveredDelegateOwnerWallet) {
+    throw new ErrorBadRequest(
+      'Health check response must be signed by delegate owner wallet registered on L1 for spID'
+    )
+  }
+
+  /**
+   * Generate proposal signature artifacts
+   */
+  const proposalSignatureInfo = await audiusLibs.contracts.UserReplicaSetManagerClient.getProposeAddOrUpdateContentNodeRequestData(
+    /* new node L1 spID */ spID,
+    /* new node L1 delegateOwnerWallet */ delegateOwnerWalletFromSPFactory,
+    /* proposing node (self) ownerWallet */ ownerWalletFromSPFactory,
+    /* proposing node (self) L1 spID */ nodeConfig.get('spID')
+  )
+
+  return {
+    nonce: proposalSignatureInfo.nonce,
+    sig: proposalSignatureInfo.sig,
+    spID: nodeConfig.get('spID')
```

Comment:
nit - why is this better than passing in the spID object you already have

Reply:
this is actually not the same - the spID object passsed into this function is the requesting node's spID, this is the proposer node's spID